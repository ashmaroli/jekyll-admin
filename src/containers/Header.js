import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router';

export class Header extends Component {

  render() {
    const { config, admin } = this.props;
    return (
      <div className="header">
        <h3 className="title">
          <Link target="_blank" to={config.url || '/'}>
            <i className="fa fa-home" />
            <span>{config.title || 'You have no title!'}</span>
          </Link>
        </h3>
        <span className="version">{admin.version}</span>
      </div>
    );
  }
}

Header.propTypes = {
  admin: PropTypes.object.isRequired,
  config: PropTypes.object.isRequired,
};

export default Header;
