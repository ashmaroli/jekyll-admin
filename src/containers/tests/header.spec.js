import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';
import { Header } from '../Header';
import { config } from './fixtures';

const defaultProps = {
  admin: { version: '0.1.0' },
  config: config
};

function setup(props) {
  const component = mount(
    <Header {...props} />
  );

  return {
    component: component,
    title: component.find('h3 span'),
  };
}

describe('Containers::Header', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Header {...defaultProps} />, div);
  });

  it('should render correctly', () => {
    const { component, title } = setup(defaultProps);
    const { config } = component.props();
    expect(title.text()).toEqual(config.title);

    const tree = renderer.create(<Header {...defaultProps} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render placeholder title', () => {
    const props = { ...defaultProps, config: {} };
    const { component, title } = setup(props);
    expect(title.text()).toEqual('You have no title!');

    const tree = renderer.create(<Header {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
