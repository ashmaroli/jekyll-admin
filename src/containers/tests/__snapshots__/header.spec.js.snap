// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Containers::Header should render correctly 1`] = `
<div
  className="header"
>
  <h3
    className="title"
  >
    <a
      onClick={[Function]}
      style={Object {}}
      target="_blank"
    >
      <i
        className="fa fa-home"
      />
      <span>
        Your awesome title
      </span>
    </a>
  </h3>
  <span
    className="version"
  >
    0.1.0
  </span>
</div>
`;

exports[`Containers::Header should render placeholder title 1`] = `
<div
  className="header"
>
  <h3
    className="title"
  >
    <a
      onClick={[Function]}
      style={Object {}}
      target="_blank"
    >
      <i
        className="fa fa-home"
      />
      <span>
        You have no title!
      </span>
    </a>
  </h3>
  <span
    className="version"
  >
    0.1.0
  </span>
</div>
`;
